user_sync/credential_manager.py

```diff
@@ -42,5 +44,16 @@ def get_credentials(credential_type, credential_id, **kwArgs):
     :type kwArgs: dict
     :rtype dict | str | None
     '''
+    if credential_type == DIRECTORY_CREDENTIAL_TYPE:
+        config = kwArgs['config']
+        if "credential_manager" in config:
+            cred_man = config['credential_manager']
+            for supported in KEYRING_SUPPORTED:
+                if cred_man['type'] == supported:
+                    return get_credentials_from_keyring(cred_man['service_name'], cred_man['username'])
```

Comment:
you need to wrap this in a try/catch and only return credentials if they were successfully fetched.  Otherwise you should log the failure as Critical and return None (so the original file values are used).  Then the ldap connector will fail to authenticate and raise an exception.